libs/ledgerjs/packages/cryptoassets/src/tokens.ts

```diff
@@ -241,8 +247,13 @@ export function convertERC20([
   contractAddress,
   disableCountervalue,
   delisted,
-]: ERC20Token | PolygonERC20Token): TokenCurrency {
-  const parentCurrency = getCryptoCurrencyById(parentCurrencyId);
+]: ERC20Token | PolygonERC20Token): TokenCurrency | undefined {
+  const parentCurrency = findCryptoCurrencyById(parentCurrencyId);
```

Comment:
Another way of doing this, a bit more concise and doesn't need to change the `convertXXXTokens` and `addTokens` functions:
```diff
-addTokens(erc20tokens.map(convertERC20));
+addTokens(erc20tokens.filter(t => !!findCryptoCurrencyById(t[0])).map(convertERC20));
```

Reply:
Like I was saying to @lvndry earlier, this would make 2 loops instead of 1 on a fairly big (7k items today) array, and it would modify our "source of truth" by not really dealing with whatever file we have locally (which doesn't really makes sense). 

I would advise against this

Reply:
I (briefly) though about perf, this would cause 1 more instruction for each processed token, so about ~30000 reads for the sake of code readability. I don't think that should be significant event on mobile?